api/src/helpers/filters/taxa_filter.rs

```diff
@@ -0,0 +1,25 @@
+use std::collections::HashSet;
+use datastore::LineageStore;
+use index::ProteinInfo;
+use crate::helpers::filters::UniprotFilter;
+use crate::helpers::lineage_helper::{get_lineage_array, LineageVersion};
+
+pub struct TaxaFilter<'a> {
+    pub taxa: HashSet<u32>,
+    lineage_store: &'a LineageStore
+}
+
+impl UniprotFilter for TaxaFilter<'_> {
+    fn filter(&self, protein: &ProteinInfo) -> bool {
+        get_lineage_array(protein.taxon, LineageVersion::V2, self.lineage_store)
+            .iter()
+            .flatten()
+            .any(|ancestor| self.taxa.contains(&(ancestor.abs() as u32)))
+    }
+}
+
+impl<'a> TaxaFilter<'a> {
+    pub fn new(taxa: HashSet<u32>, lineage_store: &'a LineageStore) -> Self {
+        TaxaFilter { taxa, lineage_store }
+    }
+}
```

Comment:
When creating a new instance of the `ProteomeFilter`, an error can be thrown if something goes wrong while retrieving information about the proteome from the UniProt API. For the taxa, maybe we should consider doing the same. In this function, we should check if all given taxa IDs are known to Unipept. If not, we need to report to the user that they provided an unknown taxon ID (using an HTTP status code 422).